packages/router/src/router.tsx

```diff
@@ -151,51 +139,109 @@ const Router: React.FC<RouterProps> = ({
   </LocationProvider>
 )
 
-const LocationAwareRouter: React.FC<RouterProps> = ({
+/**
+ * `namedRoutes` is the `routes` object that you import from `@redwoodjs/router`:
+ *
+ * ```js
+ * import { Link, routes } from '@redwoodjs/router'
+ *
+ * // ...
+ *
+ * <Link to={routes.home()}>Home</Link>
+ *
+ * // ...
+ * ```
+ *
+ * It's populated by the router, at run-time.
+ */
+const namedRoutes: AvailableRoutes = {}
+
+const LocationAwareRouter = ({
   useAuth,
   paramTypes,
   pageLoadingDelay,
   children,
-}) => {
+}: React.PropsWithChildren<RouterProps>) => {
   const location = useLocation()
+
   const flatChildArray = flattenAll(children)
 
-  const hasHomeRoute = flatChildArray.some((child) => {
-    if (isRoute(child)) {
-      return child.props.path === '/'
-    }
+  /**
+   * Validate the props of the routes in the router.
+   *
+   * Routes need to have `path`, `page`, and `name`,
+   * while redirect routes need to have `path` and `redirect`:
+   *
+   * ```js
+   * <Route path="/" page={HomePage} name="home" />  // ok
+   * <Route path="/" redirect="/home" />  // ok
+   * ```
+   */
+  flatChildArray
+    .filter((child) => isRoute(child) && !child.props.notfound)
```

Comment:
See if you can include something in the big comment above about why we're skipping notfound here